src/simplicial-hott/06-2cat-of-segal-types.rzk.md

```diff
@@ -171,30 +173,70 @@ Equivalently , natural transformations can be determined by their **components**
       is-equiv-ev-components-nat-trans A B f g)
 ```
 
-### Horizontal composition
+### Naturality square
 
-Horizontal composition of natural transformations makes sense over any type. In
-particular , contrary to what is written in [RS17] we do not need `#!rzk C` to
-be Segal.
+Natural transformations are automatically natural when the codomain is a Segal
+type.
 
-```rzk
-#def horizontal-comp-nat-trans
-  ( A B C : U)
-  ( f g : A → B)
-  ( f' g' : B → C)
-  ( η : nat-trans A (\ _ → B) f g)
-  ( η' : nat-trans B (\ _ → C) f' g')
-  : nat-trans A (\ _ → C) (\ x → f' (f x)) (\ x → g' (g x))
-  := \ t x → η' t (η t x)
+```rzk title="RS17, Proposition 6.6"
+#section comp-eq-square-is-segal
 
-#def horizontal-comp-nat-trans-components
-  ( A B C : U)
-  ( f g : A → B)
-  ( f' g' : B → C)
-  ( η : nat-trans-components A (\ _ → B) f g)
-  ( η' : nat-trans-components B (\ _ → C) f' g')
-  : nat-trans-components A (\ _ → C) (\ x → f' (f x)) (\ x → g' (g x))
-  := \ x t → η' (η x t) t
+#variable A : U
+#variable is-segal-A : is-segal A
+#variable α : (Δ¹×Δ¹) → A
+
+#def α00 : A := α (0₂,0₂)
+#def α01 : A := α (0₂,1₂)
+#def α10 : A := α (1₂,0₂)
+#def α11 : A := α (1₂,1₂)
+
+#def α0* : Δ¹ → A := \ t → α (0₂,t)
+#def α1* : Δ¹ → A := \ t → α (1₂,t)
+#def α*0 : Δ¹ → A := \ s → α (s,0₂)
+#def α*1 : Δ¹ → A := \ s → α (s,1₂)
+#def α-diag : Δ¹ → A := \ s → α (s,s)
+
+#def lhs uses (α) : Δ¹ → A := comp-is-segal A is-segal-A α00 α01 α11 α0* α*1
+#def rhs uses (α) : Δ¹ → A := comp-is-segal A is-segal-A α00 α10 α11 α*0 α1*
```

Comment:
These will become `#let` when it's implemented (see https://github.com/rzk-lang/rzk/issues/92). Nothing to fix for now, just making a note.